src/modules/job-info/job_state.c

```diff
@@ -1364,6 +1437,19 @@ static struct job *eventlog_restart_parse (struct info_ctx *ctx,
                 update_job_state (ctx, job, FLUX_JOB_CLEANUP, timestamp);
         }
         else if (!strcmp (name, "alloc")) {
+            /* context not required if no annotations */
+            if (context) {
+                json_t *annotations;
+                if (json_unpack (context,
+                                 "{ s:o }",
+                                 "annotations", &annotations) < 0) {
+                    flux_log_error (ctx->h, "%s: alloc context for %ju invalid",
```

Comment:
errno not set by `json_unpack()` but `flux_log_error()` is called.

also would json_object_get() be a lighter weight equivalent to `json_unpack ("s:o")`?